jina/executors/indexers/__init__.py

```diff
@@ -215,6 +215,12 @@ def query(self, key: Any) -> Optional[Any]:
         """
         raise NotImplementedError
 
+    def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
+        raise NotImplementedError
+
+    def delete(self, keys: Iterator[int], *args, **kwargs):
+        raise NotImplementedError
```

Comment:
curious, why not `__setitem__` and `__del__`?

Reply:
Nice catch! Actually, I don't see any argument for not using them. Changing this could be done in another ticket.
@maximilianwerk Do you know why we don't use dunder methods here?

Reply:
Currently, `__getitem__` is just calling `query` in the `BaseKVIndexer`. Following this approach, we would need to implement `__setitem__` with either calling `add` or `update`. Since we bind the CRUD operations even to dedicated request objects, it would be weird to first make an explicit update call, and then convert it to a `__setitem__` which needs to figure out again, whether it is a `add` or `update`. Anyhow, for convenience, we should add the `__del__` and `__setitem__` later on.